Conslidate normative lanugage on x5t and x5chain usage (plus SecConSec content) (#241)
ietf-wg-scitt/draft-ietf-scitt-architecture: draft-ietf-scitt-architecture.md

**Editor's Note:**: *The label "394" is expected to be reserved by this document, in the COSE Header Parameters Registry.*

The terms "header", "payload", and "to-be-signed bytes" are defined in {{-COSE}}.

> The term "claim" is defined in {{RFC8392}} and is repeated here for readability:

Reviewer: We also still have the term "Claim" in [the terminology section](https://ietf-wg-scitt.github.io/draft-ietf-scitt-architecture/x5chain-usage/draft-ietf-scitt-architecture.html#section-2). Should we not remove that and replace it with this wording? Because of his git diffs work I cannot recommend that here, or I would do it for you, but this seems like an important change to not cause confusion down the road.

Reviewer 2: Agreed to remove claims in the terminology section.

Once all the Envelope headers are set, an Issuer MUST use a standard COSE implementation to produce an appropriately serialized Signed Statement.
The SCITT tag `COSE_Sign1_Tagged` is outside the scope of COSE, and used to indicate that a signed object is a Signed Statement.

> Issuers MAY produce Signed Statements about different Artifacts under the same Identity.

Reviewer 3: This is a throwaway, consider just removing this normative statement.

Reviewer 2: Keeping this line puts context to how multiple Signed Signed Statements can be used. 
Changing to a colocial "can" to keep the description, without making it vague normative language. 
```suggestion
Issuers can produce Signed Statements about different Artifacts under the same Identity.
```

For example, if a Transparency Service is implemented using a set of redundant replicas, each running within its own hardware-protected trusted execution environments (TEEs), then each replica can provide fresh Evidence or fresh Attestation Results about its TEEs. The respective Evidence can show, for example, the binding of the hardware platform to the software that runs the Transparency Service, the long-term public key of the service, or the key used by the replica for signing Receipts. The respective Attestation Result, for example, can show that the remote attestation Evidence was appraised by a trusted Verifier and complies with well-known Reference Values and Endorsements.

Auditors should be aware that the certification path information included in an unprotected `x5chain` header of a to-be-registered Signed Statement can be tampered with by a malicious Transparency Service (e.g., one that does not incorporate remote attestation), which may replace the intermediate certificates and ultimately connect to an unexpected root.
This modification can allow malicious TS to forge Claims that look genuine except for the wrong trust anchor.

> In addition to performing chain validation in accordance with PKIX rules specified in {{-PKIX}}, auditors should explicitly verify the root certificates.

Reviewer 4: ```suggestion
Auditors MUST perform certification path validation in accordance with PKIX rules specified in {{-PKIX}}, In particular, auditors MUST verify that certification paths chain to a trusted root certificate.
```

> In addition to performing chain validation in accordance with PKIX rules specified in {{-PKIX}}, auditors should explicitly verify the root certificates are genuine.

Reviewer: Genuine sounds subjective.

```suggestion
In addition to performing chain validation in accordance with PKIX rules specified in {{-PKIX}}, auditors should explicitly verify the root certificates.
```

Author: Not much lost here. I'd be okay with that suggestion.

Auditors should be aware that the certification path information included in an unprotected `x5chain` header of a to-be-registered Signed Statement can be tampered with by a malicious Transparency Service (e.g., one that does not incorporate remote attestation), which may replace the intermediate certificates and ultimately connect to an unexpected root.
This modification can allow malicious TS to forge Claims that look genuine except for the wrong trust anchor.
Auditors MUST perform certification path validation in accordance with PKIX rules specified in {{-PKIX}}.

> In particular, auditors MUST verify that certification paths chain to one or more trust anchors (often represented as root certificates).

Reviewer 2: Editors call: agreeing to the content, question if this is the correct section, which we can move in another PR.
